[18.0][IMP] web_refresher: add auto refresh on interval

O-Mutt opened the pull request:
This adds a new "auto" refresh option to the UI for allowing an interval refresh. I have also implemented a setting that is stored in localStorage for the auto refresh interval. When the auto refresh is set it puts the value into localStorage to be re-used on each screen by default.

---

### Here is the auto refresh in action with network traffic included:

https://github.com/user-attachments/assets/49b25b73-3c6b-4564-a17b-b57b3bf93931

---

### UI of baseline
<img width="543" height="203" alt="Screenshot 2025-08-21 at 1 42 31 PM" src="https://github.com/user-attachments/assets/f91e8206-2c58-4134-b346-0e86f710c2fa" />

---

### UI w/hover
<img width="572" height="213" alt="Screenshot 2025-08-21 at 1 42 35 PM" src="https://github.com/user-attachments/assets/72557ddb-638f-4fc6-bc65-05997546ef06" />

---

### UI Clicked dropdown
<img width="517" height="340" alt="Screenshot 2025-08-21 at 1 42 39 PM" src="https://github.com/user-attachments/assets/ba55c697-a853-4b05-8b95-2fb9c97c56a5" />

---

### UI showing animation of manual refresh button to indicate ongoing action
<img width="421" height="219" alt="Screenshot 2025-08-21 at 1 42 42 PM" src="https://github.com/user-attachments/assets/57293441-0431-40c6-989d-a878447b72af" />

SilvioC2C committed:
[14.0] web_m2x_options_manager: New module to manage create/edit options from interface

Tardo committed:
[ADD] web_time_range_menu_custom

Tardo committed:
[FIX] web_time_range_menu_custom: Mix custom and 'no custom' periods

Steps to reproduce:

- Go to pivot view
- Select a custom time range
- Select a predefined comparison period (Ex. Previous Period)

Now you see that the comparison period is not being applied.
After this commit the comparison period is applied correctly.

TT29249

oca-travis committed:
[UPD] Update web_m2x_options_manager.pot

OCA-git-bot committed:
[UPD] README.rst

yankinmax committed:
[UPD] web_m2x_options_manager: Add option to prevent Create/Edit wizard to pop up

oca-travis committed:
[UPD] Update web_m2x_options_manager.pot

OCA-git-bot committed:
web_m2x_options_manager 14.0.1.1.0

fkantelberg committed:
[ADD][16.0] Module web_dark_mode

Commit:
[UPD] Update web_dark_mode.pot

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
[ADD] icon.png

GoodERPJeff committed:
Added translation using Weblate (Chinese (zh))

ibuioli committed:
Added translation using Weblate (Spanish (Argentina))

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (4 of 4 strings)

Translation: web-16.0/web-16.0-web_dark_mode
Translate-URL: https://translation.odoo-community.org/projects/web-16-0/web-16-0-web_dark_mode/es_AR/

CarlosRoca13 committed:
[IMP] web_time_range_menu_custom: black, isort, prettier

fkantelberg committed:
[FIX] web_dark_mode: Fix uninstallation

OCA-git-bot committed:
web_dark_mode 16.0.1.0.1

badbole committed:
Added translation using Weblate (Croatian)

badbole committed:
Translated using Weblate (Croatian)

Currently translated at 100.0% (4 of 4 strings)

Translation: web-16.0/web-16.0-web_dark_mode
Translate-URL: https://translation.odoo-community.org/projects/web-16-0/web-16-0-web_dark_mode/hr/

tate11 committed:
Added translation using Weblate (Turkish)

tate11 committed:
Translated using Weblate (Turkish)

Currently translated at 100.0% (4 of 4 strings)

Translation: web-16.0/web-16.0-web_dark_mode
Translate-URL: https://translation.odoo-community.org/projects/web-16-0/web-16-0-web_dark_mode/tr/

francesco-ooops committed:
Added translation using Weblate (Italian)

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 76.4% (26 of 34 strings)

Translation: web-14.0/web-14.0-web_m2x_options_manager
Translate-URL: https://translation.odoo-community.org/projects/web-14-0/web-14-0-web_m2x_options_manager/it/

CarlosRoca13 committed:
[MIG] web_time_range_menu_custom: Migration to 15.0

CarlosRoca13 committed:
[IMP] web_time_range_menu_custom: Dates include today but other granularities not

Ivorra78 committed:
Added translation using Weblate (Spanish)

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (4 of 4 strings)

Translation: web-16.0/web-16.0-web_dark_mode
Translate-URL: https://translation.odoo-community.org/projects/web-16-0/web-16-0-web_dark_mode/es/

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
[UPD] README.rst

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: web-14.0/web-14.0-web_m2x_options_manager
Translate-URL: https://translation.odoo-community.org/projects/web-14-0/web-14-0-web_m2x_options_manager/

gregory-moka committed:
Added translation using Weblate (French)

gregory-moka committed:
Translated using Weblate (French)

Currently translated at 50.0% (2 of 4 strings)

Translation: web-16.0/web-16.0-web_dark_mode
Translate-URL: https://translation.odoo-community.org/projects/web-16-0/web-16-0-web_dark_mode/fr/

HekkiMelody committed:
[FIX] all: fix pre-commit new checks

Commit:
[UPD] Update web_m2x_options_manager.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: web-14.0/web-14.0-web_m2x_options_manager
Translate-URL: https://translation.odoo-community.org/projects/web-14-0/web-14-0-web_m2x_options_manager/

fkantelberg committed:
[FIX] web_dark_mode: Add fields to field lists

OCA-git-bot committed:
[BOT] post-merge updates

Ivorra78 committed:
Added translation using Weblate (Spanish)

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (4 of 4 strings)

Translation: web-16.0/web-16.0-web_dark_mode
Translate-URL: https://translation.odoo-community.org/projects/web-16-0/web-16-0-web_dark_mode/it/

yajo committed:
[ADD] web_touchscreen: enable kanban sub-form views for tablets

This module prefers kanban sub-form views when using a tablet, or any other
device with a touch screen, no matter its size.

@moduon MT-4472

Commit:
[UPD] Update web_touchscreen.pot

yajo committed:
[DOCS] web_touchscreen: add screenshot

Follow suggestion from https://github.com/OCA/web/pull/2701#issuecomment-1870681682.

@moduon MT-4472

OCA-git-bot committed:
[BOT] post-merge updates

mymage committed:
Added translation using Weblate (Italian)

CarlosRoca13 committed:
[MIG] web_time_range_menu_custom: Migration to 16.0

mymage committed:
Added translation using Weblate (Italian)

CarlosRoca13 committed:
[IMP] web_time_range_menu_custom: pre-commit auto fixes

CarlosRoca13 committed:
[MIG] web_time_range_menu_custom: Migration to 17.0

CarlosRoca13 committed:
[IMP] web_time_range_menu_custom: pre-commit auto fixes

CarlosRoca13 committed:
[MIG] web_time_range_menu_custom: Migration to 18.0

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (33 of 33 strings)

Translation: web-14.0/web-14.0-web_m2x_options_manager
Translate-URL: https://translation.odoo-community.org/projects/web-14-0/web-14-0-web_m2x_options_manager/it/

houzefa-abba committed:
[IMP] web_timeline: Safer demo view action override

Append our view_mode this way to avoid overwriting changes done by other
modules.

Chionne27 committed:
[FIX] web_m2x_options_manager: writing mode when create/edit wizard is false

OCA-git-bot committed:
[BOT] post-merge updates

ivantodorovich committed:
[ADD] web_widget_popover

MohamedOsman7 committed:
[IMP] web_chatter_position: improve form and chatter layout styles

Co-authored-by: Stefan Rijnhart <stefan@opener.amsterdam>

anusriNPS committed:
[16.0][ADD] web_sort_menu: Module web_sort_menu

anusriNPS committed:
[REF] web_company_color: Update fetching scss_template

   Using a function to fetch SCSS_TEMPLATE

O-Mutt committed:
[IMP] web_refresher: add auto refresh on interval

O-Mutt committed:
[IMP] web_refresher: add auto refresh on interval

O-Mutt committed:
[FIX] web_refresher: linting

O-Mutt committed:
[FIX] web_refresher: linting

O-Mutt committed:
[CHORE] web_refresher: one more lint fix

O-Mutt committed:
[CHORE] web_refresher: one more lint fix

O-Mutt commented on web_refresher/static/src/xml/refresher.xml:
```diff
@@ -5,13 +5,61 @@
+                    title="Refresh"
+                    tabindex="-1"
+                >
+                    <i id="manual-refresh-icon" class="fa fa-refresh" />
```
Moved the icons to an `<i>` to allow the `fa-spin` to be added and removed without spinning the whole button 🤣

O-Mutt replied:
added `id` for easy selections

O-Mutt commented on web_refresher/static/src/xml/refresher.xml:
```diff
@@ -5,13 +5,61 @@
+                    id="manual-refresh-btn"
+                    class="btn btn-secondary m-0"
+                    aria-label="Refresh"
+                    t-on-click="onClickRefresh"
```
Maintain existing flow with the a reset of the new timeout when manually clicked

O-Mutt commented on web_refresher/static/src/xml/refresher.xml:
```diff
@@ -5,13 +5,61 @@
+                            t-on-click="onChangeAutoRefreshInterval"
+                            value="-1"
+                        >Off</button>
+                        <button
+                            class="dropdown-item"
+                            t-on-click="onChangeAutoRefreshInterval"
+                            value="1000"
+                        >1s</button>
```
I'm not sure 1s is really appropriate. I could easily be persuaded to remove `1s`

O-Mutt commented on web_refresher/static/src/xml/refresher.xml:
```diff
@@ -5,13 +5,61 @@
+                        Auto Refresh: Off
+                    </button>
+                    <div class="dropdown-menu" aria-labelledby="auto-refresh-dd">
+                        <button
+                            class="dropdown-item"
+                            t-on-click="onChangeAutoRefreshInterval"
+                            value="-1"
+                        >Off</button>
```
Used `-1` as the "off" value to avoid `null`/`undefined` confusions. #javascriptthings

O-Mutt commented on web_refresher/static/src/xml/refresher.xml:
```diff
@@ -5,13 +5,61 @@
+                        data-bs-toggle="dropdown"
+                        aria-expanded="false"
+                    >
+                        Auto Refresh: Off
```
Default text could be moved out to allow easier i18n

O-Mutt commented on web_refresher/static/src/xml/refresher.xml:
```diff
@@ -5,13 +5,61 @@
+            <div class="btn-group" role="group">
+                <div class="btn-group" role="group">
+                    <button
+                        id="auto-refresh-dd"
```
added `id` for easy selections

O-Mutt commented on web_refresher/static/src/js/refresher.esm.js:
```diff
@@ -111,6 +124,35 @@ export class Refresher extends Component {
+            } else {
+                intervalValue = `${intervalInSeconds}s`;
+            }
+            document.getElementById("manual-refresh-icon").classList.add("fa-spin");
```
add `fa-spin` to show active auto refresh

O-Mutt commented on web_refresher/static/src/js/refresher.esm.js:
```diff
@@ -111,6 +124,35 @@ export class Refresher extends Component {
+        } else {
+            const intervalInSeconds = this.refreshInterval / 1000;
+            if (intervalInSeconds >= 60) {
+                intervalValue = `${Math.floor(intervalInSeconds / 60)}min`;
```
60s or more gets "min" added

O-Mutt commented on web_refresher/static/src/js/refresher.esm.js:
```diff
@@ -111,6 +124,35 @@ export class Refresher extends Component {
+        }
+        document.getElementById("auto-refresh-dd").textContent =
+            `Auto Refresh: ${intervalValue}`;
+    }
```
Reviewing this logic i don't love how it played out. I'm going to refactor

O-Mutt commented on web_refresher/static/src/js/refresher.esm.js:
```diff
@@ -81,6 +84,16 @@ export class Refresher extends Component {
         if (!updated) {
             updated = this._searchModelRefresh();
         }
+        // Check the refreshInterval is greater than 0 and start a timer for the next refresh
+        if (this.refreshInterval > 0) {
+            // Always attempt to clear a running timeout in case the refresh was done manually
+            if (typeof this.runningRefresherId === "number") {
+                clearTimeout(this.runningRefresherId);
+            }
+            this.runningRefresherId = setTimeout(() => {
+                this.refresh();
+            }, this.refreshInterval);
+        }
```
Consolidated logic for starting the refresh interval in the current refresh to allow recursive calls

O-Mutt committed:
[CHORE] lint

O-Mutt committed:
[CHORE] lint

O-Mutt commented on web_refresher/static/src/js/refresher.esm.js:
```diff
@@ -111,6 +124,29 @@ export class Refresher extends Component {
+    onChangeAutoRefreshInterval(clickedOption) {
+        const newInterval =
+            parseInt(clickedOption.value ?? clickedOption.target.value) ?? -1;
+        const newIntervalText =
+            clickedOption.textContent ?? clickedOption.target.textContent;
```
Use the xml button text as the interval on the label

O-Mutt commented on web_refresher/__manifest__.py:
```diff
@@ -1,6 +1,6 @@
 {
     "name": "Web Refresher",
-    "version": "18.0.1.0.0",
+    "version": "18.0.2.0.0",
```
```suggestion
    "version": "18.1.1.0.0",
```

O-Mutt commented on web_refresher/__manifest__.py:
```diff
@@ -1,6 +1,6 @@
 {
     "name": "Web Refresher",
-    "version": "18.0.1.0.0",
+    "version": "18.1.1.0.0",
```
```suggestion
    "version": "18.0.2.0.0",
```

sergioM-S73 committed:
fix error: env.services.rpc is not a function

len-foss committed:
[MIG] web_dark_mode: Migration to 18.0

len-foss committed:
[IMP] web_dark_mode: pre-commit execution

O-Mutt committed:
[CHORE] web_refresher: one more lint fix

O-Mutt committed:
[CHORE] web_refresher: one more lint fix

O-Mutt committed:
[IMP] web_refresher: make the interval text cleaner

O-Mutt committed:
[IMP] web_refresher: make the interval text cleaner

O-Mutt committed:
[FIX] parse int from localstorage

O-Mutt committed:
[FIX] parse int from localstorage

O-Mutt committed:
[FIX] Moving state to local storage and defaulting the value on load

O-Mutt committed:
[FIX] Moving state to local storage and defaulting the value on load

O-Mutt committed:
[IMP] Slightly more elegent form of loading

O-Mutt committed:
[IMP] Slightly more elegent form of loading

O-Mutt committed:
[LINT] fix lint errors

O-Mutt committed:
[LINT] fix lint errors

O-Mutt committed:
[LINT] Again

O-Mutt committed:
[LINT] Again

O-Mutt committed:
[IMP] add per page settings and a default

O-Mutt committed:
[IMP] add per page settings and a default

O-Mutt committed:
[FIX] Fixing up the "default" setting when not yet set

O-Mutt committed:
[FIX] Fixing up the "default" setting when not yet set

O-Mutt committed:
[LINT] Always with the lint failures

O-Mutt committed:
[LINT] Always with the lint failures

yostashiro committed:
[ADD] web_form_banner

AungKoKoLin1997 committed:
[IMP] web_form_banner: black, isort, prettier

AungKoKoLin1997 committed:
[MIG] web_form_banner: Migration to 15.0

Refactor JS:
- Switch to the ESM style
- Change var to const/let
- Remove unused parts (diff, el.dataset.wfbTriggerFields)

Miscellaneous improvements:
- list view optional
- Switch archive box icon to banner

Co-authored-by: Yoshi Tashiro <tashiro@quartile.co>

DavidJForgeFlow committed:
[IMP] web_chatter_position: pre-commit fixes.

AungKoKoLin1997 committed:
[IMP] web_form_banner: pre-commit auto fixes

AungKoKoLin1997 committed:
[MIG] web_form_banner: Migration to 16.0

yostashiro committed:
[IMP] web_form_banner: Partially support in-group target XPath

- Add (partial) support for elements inside groups to be a target
  XPath without distorting the presentation
- Remove obsolete bits

ljmnoonan committed:
[IMP] web_dark_mode Set dark theme to immitate VSCode Modern Dark
Also, change asset bundle loading from 'prepend' to 'before' This is in line with Odoo's own way of doing it.

ljmnoonan committed:
[IMP] web_dark_mode Change navbar from purple to dark
Also get rid of the bottom border.

ljmnoonan committed:
[IMP] web_dark_mode Fix tint and shade color functions

AungKoKoLin1997 committed:
[IMP] web_form_banner: pre-commit auto fixes

AungKoKoLin1997 committed:
[MIG] web_form_banner: Migration to 17.0

ljmnoonan committed:
[IMP] web_dark_mode Add lazy loading asset bundle
If you go to any activity view and hover over an empty cell you will notice that
without this fix the the on hover background of empty cells (.o_activity_empty_cell) set by
$component-active-bg which is itself set as mix($o-action, $o-gray-100, 20%) through $o-component-active-bg
completely ignores the overrides we set in primary_variables.dark.scss.
This is because the assets for that view are lazy loaded, so we need to include our override
bundles in the lazy bundle.

ljmnoonan committed:
[IMP] web_dark_mode Override BS variables
Just general customizations of BS variables to bring them in line with our style

ljmnoonan committed:
[IMP] web_dark_mode Improve notebook tab styling

AungKoKoLin1997 committed:
[IMP] web_form_banner: pre-commit auto fixes

AungKoKoLin1997 committed:
[MIG] web_form_banner: Migration to 18.0

AungKoKoLin1997 committed:
[IMP] web_form_banner: imp readme

ljmnoonan committed:
[IMP] web_dark_mode Adjust tag and colorlist colors

ljmnoonan committed:
[IMP] web_dark_mode Adjust calendar coloring
Add secondary_variables override to get rid of colors that don't jive well with the dark background.
Unfortunately this does introduce an issue where if you have many calendar users, you can run out of colors
This issue exists in the enterprise dark mode as well, so hopefully it will be fixed.

ljmnoonan committed:
[IMP] web_dark_mode Adjust the styles of custom properties

ljmnoonan committed:
[IMP] web_dark_mode Tweak the search bar
Set the background color to be darker and tone down the border around it.
The styling for the border is borrwed from the style of .o-mail-Composer-bg

ljmnoonan committed:
[IMP] web_dark_mode Set image background to white

ljmnoonan committed:
[IMP] web_dark_mode Tweak kanban controller
Set attachment image background to white and fix the column background when hovering with a draggable card

AungKoKoLin1997 committed:
[FIX] web_form_banner

- Fix not updating banner when pager is update or duplicate or create
- Fix presentation of help in web form banner rule

ljmnoonan committed:
[IMP] web_theme_classic Refactor SCSS
Refactor all selectors to mirror the way odoo sets them. This results in a MUCH broader application of our styles that is not limited to form views.
Prefix all vars with "wtc-" in order to prevent them from overriding or being overridden by bootstrap vars.
Suffix all vars with !default to ensure that they can be overriden by dark mode.
Add $wtc-input-color-required to ensure text contrasts on $wtc-input-background-color-required. This is especially necessary in instances like the quantity and unit price columns of a sale order because by default these fields are set to a light blue.
Remove border on .note-editable, i.e. the HTML editor. This feature was broken and, as far as I am concerned, unnecessary.
Remove all button customization as it was all broken and is superceeded by OCA/web/web_save_discard_button anyways.

ljmnoonan committed:
[IMP] web_theme_classic Add dark mode assets
web_dark_mode is not merged yet.
https://github.com/OCA/web/pull/3324

ljmnoonan committed:
[IMP] web_dark_mode: Add myself to CONTRIBUTORS.md
I don't want to but my email on a public repo, so I just linked my github page instead.
I could not find any guideline against this, but please correct me if there is.

yankinmax committed:
[IMP] web_widget_popover: pre-commit auto fixes

yankinmax committed:
[18.0][MIG] web_widget_popover: Migration to 18.0

O-Mutt committed:
fix: memory leak due to held refrence blocking GC

O-Mutt committed:
fix: memory leak due to held refrence blocking GC

O-Mutt committed:
Hide on small view auto refresh

O-Mutt committed:
Hide on small view auto refresh

ljmnoonan committed:
[IMP] web_dark_mode Change thead active (column filter) bg for lists in forms
It was being set to $o-form-lightsecondary which I set lighter, so this tones it down a bit.

OCA-git-bot committed:
[BOT] post-merge updates

Commit:
[UPD] Update web_form_banner.pot

OCA-git-bot committed:
[BOT] post-merge updates

LauraCForgeFlow committed:
[ADD] web_widget_product_label_section_and_note_full_label

mymage committed:
Added translation using Weblate (Italian)

anusriNPS committed:
[MIG] web_sort_menu: Migration to 18.0

Commit:
[UPD] Update web_widget_popover.pot

OCA-git-bot committed:
[BOT] post-merge updates

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (63 of 63 strings)

Translation: web-18.0/web-18.0-web_form_banner
Translate-URL: https://translation.odoo-community.org/projects/web-18-0/web-18-0-web_form_banner/it/

primes2h committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (31 of 31 strings)

Translation: web-18.0/web-18.0-web_responsive
Translate-URL: https://translation.odoo-community.org/projects/web-18-0/web-18-0-web_responsive/it/

SilvioC2C committed:
[IMP] web_m2x_options_manager: pre-commit auto fixes

SilvioC2C committed:
[MIG] web_m2x_options_manager: Migration to 17.0

Field ``option_create_edit_wizard`` has been dropped in the migration: the pop-up wizard does not exist anymore, so the field has no use

OCA-git-bot committed:
[BOT] post-merge updates

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (3 of 3 strings)

Translation: web-18.0/web-18.0-web_widget_popover
Translate-URL: https://translation.odoo-community.org/projects/web-18-0/web-18-0-web_widget_popover/it/

Ryoko04 committed:
Added translation using Weblate (Japanese)

Ryoko04 committed:
Translated using Weblate (Japanese)

Currently translated at 100.0% (2 of 2 strings)

Translation: web-18.0/web-18.0-web_m2x_options
Translate-URL: https://translation.odoo-community.org/projects/web-18-0/web-18-0-web_m2x_options/ja/

OCA-git-bot committed:
[BOT] post-merge updates

Commit:
[UPD] Update web_chatter_position.pot

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: web-18.0/web-18.0-web_chatter_position
Translate-URL: https://translation.odoo-community.org/projects/web-18-0/web-18-0-web_chatter_position/

Xino61122 committed:
[IMP] web_touchscreen: pre-commit auto fixes

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (5 of 5 strings)

Translation: web-18.0/web-18.0-web_chatter_position
Translate-URL: https://translation.odoo-community.org/projects/web-18-0/web-18-0-web_chatter_position/it/

Commit:
[UPD] Update web_widget_product_label_section_and_note_full_label.pot

OCA-git-bot committed:
[BOT] post-merge updates

Xino61122 committed:
[MIG][18.0] web_touchscreen: Migration to 18

MT-11823

Commit:
[UPD] Update web_m2x_options_manager.pot

OCA-git-bot committed:
[BOT] post-merge updates

mymage committed:
Added translation using Weblate (Italian)

OCA-git-bot committed:
[BOT] post-merge updates

SilvioC2C committed:
[IMP] web_m2x_options_manager: pre-commit auto fixes

SilvioC2C committed:
[MIG] web_m2x_options_manager: Migration to 18.0

Commit:
[UPD] Update web_m2x_options_manager.pot

OCA-git-bot committed:
[BOT] post-merge updates

bo-eska committed:
Added translation using Weblate (Turkish)

Commit:
[UPD] Update web_touchscreen.pot

OCA-git-bot committed:
[BOT] post-merge updates

Saran440 committed:
[FIX] web_save_discard_button: readme

OCA-git-bot committed:
[BOT] post-merge updates

AungKoKoLin1997 committed:
[IMP] web_form_banner: adj to resolve eslint warnings

Ricardoalso committed:
[FIX] web_m2x_options: allow field-level overrides of the global `create` and `create_edit` options

Allow field-level overrides for Many2One and Many2Many fields, while always respecting security permissions. This provides more granular control over the "Create..." and "Create and Edit..." dropdown entries without compromising access control.

Commit:
[UPD] Update web_sort_menu.pot

OCA-git-bot committed:
[BOT] post-merge updates

Commit:
[UPD] Update web_dark_mode.pot

OCA-git-bot committed:
[BOT] post-merge updates

mymage committed:
Added translation using Weblate (Italian)

RicardCForgeFlow committed:
[18.0][FIX] web_widget_x2many_2d_matrix: fix sticky header/column overlap

OCA-git-bot committed:
[BOT] post-merge updates

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (21 of 21 strings)

Translation: web-18.0/web-18.0-web_pivot_computed_measure
Translate-URL: https://translation.odoo-community.org/projects/web-18-0/web-18-0-web_pivot_computed_measure/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (8 of 8 strings)

Translation: web-18.0/web-18.0-web_widget_x2many_2d_matrix
Translate-URL: https://translation.odoo-community.org/projects/web-18-0/web-18-0-web_widget_x2many_2d_matrix/it/

LauraCForgeFlow committed:
[IMP] web_widget_one2many_tree_line_duplicate: make duplicate icon hideable

LauraCForgeFlow committed:
[ADD] web_tree_column_keyboard_resize

RicardCForgeFlow committed:
[18.0][FIX] web_widget_x2many_2d_matrix: fix total updates

Totals were not consistently refreshing immediately on cell edits.

- Replace 'onWillUpdateProps' for 'onWillRender'.
  This ensures totals update immediately when a cell value changes.
- Expand the JS test with a case verifying totals after
  multiple consecutive edits.
- Refine JS test filtering in Python HttpCase to target only this module.

Commit:
[UPD] Update web_tree_column_keyboard_resize.pot

OCA-git-bot committed:
[BOT] post-merge updates

zamberjo commented:
Hi @O-Mutt , thank you for your work on this PR.

I wanted to suggest taking a look at this other PR https://github.com/OCA/web/pull/3396. Although it targets version 17.0, it proposes an interesting approach using the Odoo Bus to update the view reactively (listening to create/write events) instead of using an automatic refresh interval (polling).
You might find it interesting as an alternative, as this method can be more efficient by avoiding unnecessary page reloads when there are no data changes.

Commit:
[UPD] Update web_widget_one2many_tree_line_duplicate.pot

OCA-git-bot committed:
[BOT] post-merge updates

mymage committed:
Added translation using Weblate (Italian)

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: web-18.0/web-18.0-web_widget_one2many_tree_line_duplicate
Translate-URL: https://translation.odoo-community.org/projects/web-18-0/web-18-0-web_widget_one2many_tree_line_duplicate/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (2 of 2 strings)

Translation: web-18.0/web-18.0-web_widget_one2many_tree_line_duplicate
Translate-URL: https://translation.odoo-community.org/projects/web-18-0/web-18-0-web_widget_one2many_tree_line_duplicate/it/

OCA-git-bot committed:
[BOT] post-merge updates

hbrunn committed:
[IMP] update dotfiles

OCA-git-bot committed:
[BOT] post-merge updates

carlos-lopez-tecnativa committed:
[FIX] web_timeline: Render the SVG correctly to draw dependencies properly

This issue was introduced during the migration to v18 when converting old jQuery code to native JavaScript in this commit: https://github.com/OCA/web/commit/258a5a92dc85fa5cc6dd9181ca444e3b4b290d0f

Fixes https://github.com/OCA/web/issues/3420

OCA-git-bot committed:
[BOT] post-merge updates

SirPyTech committed:
[IMP] web_company_color: Compile SCSS

Otherwise function calls are not evaluated in the created attachment

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
[BOT] post-merge updates

O-Mutt commented:
Honest talk that solution looks pretty slick. Would love to see it come through

> Hi @O-Mutt , thank you for your work on this PR.
> 
> I wanted to suggest taking a look at this other PR https://github.com/OCA/web/pull/3396. Although it targets version 17.0, it proposes an interesting approach using the Odoo Bus to update the view reactively (listening to create/write events) instead of using an automatic refresh interval (polling).
> You might find it interesting as an alternative, as this method can be more efficient by avoiding unnecessary page reloads when there are no data changes.

remi-filament committed:
[18.0][FIX] web_widget_x2many_2d_matrix: monetary field display

remi-filament committed:
[IMP] hide input border on monetary fields

ZubikIT committed:
Translated using Weblate (Russian)

Currently translated at 100.0% (31 of 31 strings)

Translation: web-18.0/web-18.0-web_responsive
Translate-URL: https://translation.odoo-community.org/projects/web-18-0/web-18-0-web_responsive/ru/

OCA-git-bot committed:
[BOT] post-merge updates

SirPyTech committed:
[IMP] web_timeline: Allow to select in action's view_ids

Commit:
[UPD] Update web_time_range_menu_custom.pot

OCA-git-bot committed:
[BOT] post-merge updates

arch-fan committed:
[ADD] web_toggle_chatter

Commit:
[UPD] Update web_toggle_chatter.pot

OCA-git-bot committed:
[BOT] post-merge updates

bo-eska committed:
Added translation using Weblate (Turkish)

JoanSForgeFlow committed:
[IMP] web_chatter_position: add toggle button for chatter position

mymage committed:
Added translation using Weblate (Italian)

SirPyTech committed:
[REM] web_timeline: Maintainer

OCA-git-bot committed:
[BOT] post-merge updates

dmoranp committed:
[FIX] web_responsive: remove invalid hotkey prop from DropdownItem

In Odoo 18.0, the DropdownItem component no longer accepts a hotkey
prop. Passing it causes an OWL crash:

  Invalid props for component 'DropdownItem': unknown key 'hotkey'

This error appears in mobile mode when accessing form views that have
status bar action buttons (e.g. scheduled actions).

The template extension only added the invalid hotkey attribute and had
no other content, so the file is removed entirely.

Fixes https://github.com/OCA/web/issues/3467

OCA-git-bot committed:
[BOT] post-merge updates

bo-eska committed:
Added translation using Weblate (Turkish)

OCA-git-bot committed:
[BOT] post-merge updates

hbrunn committed:
[REF] web_company_color: simplify write code

hbrunn committed:
[IMP] web_company_color: allow to set bottom border color of navbar

hbrunn committed:
[IMP] web_company_color: test ignore_company_color context key

bo-eska committed:
Translated using Weblate (Turkish)

Currently translated at 100.0% (19 of 19 strings)

Translation: web-18.0/web-18.0-web_notify
Translate-URL: https://translation.odoo-community.org/projects/web-18-0/web-18-0-web_notify/tr/

Commit:
[UPD] Update web_company_color.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: web-18.0/web-18.0-web_company_color
Translate-URL: https://translation.odoo-community.org/projects/web-18-0/web-18-0-web_company_color/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (18 of 18 strings)

Translation: web-18.0/web-18.0-web_company_color
Translate-URL: https://translation.odoo-community.org/projects/web-18-0/web-18-0-web_company_color/it/

StefanRijnhart committed:
[FIX] checklog-odoo.cfg: ignore Killing chrome descendants-or-self warning

Warning introduced in https://github.com/odoo/odoo/commit/d22e0b4e181de0940810b0f0a760bcfe288042a7. The previous warning apparently did not occur, as it was not featured in checklog-odoo.cfg.

```
2026-04-13 03:20:41,702 367 WARNING odoo odoo.addons.web_disable_export_group.tests.test_tour.TestTour.test_admin: Killing chrome descendants-or-self of 382: 6 remaining
- chrome (zombie)
- chrome (zombie)
- chrome (zombie)
- chrome (zombie)
- chrome (zombie)
- chrome (zombie)
```

Ed-Spain committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (5 of 5 strings)

Translation: web-18.0/web-18.0-web_chatter_position
Translate-URL: https://translation.odoo-community.org/projects/web-18-0/web-18-0-web_chatter_position/es/

Ed-Spain committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (3 of 3 strings)

Translation: web-18.0/web-18.0-web_widget_dropdown_dynamic
Translate-URL: https://translation.odoo-community.org/projects/web-18-0/web-18-0-web_widget_dropdown_dynamic/es/

Ed-Spain committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (8 of 8 strings)

Translation: web-18.0/web-18.0-web_widget_x2many_2d_matrix
Translate-URL: https://translation.odoo-community.org/projects/web-18-0/web-18-0-web_widget_x2many_2d_matrix/es/

Ed-Spain committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (19 of 19 strings)

Translation: web-18.0/web-18.0-web_notify
Translate-URL: https://translation.odoo-community.org/projects/web-18-0/web-18-0-web_notify/es/

Ed-Spain committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (16 of 16 strings)

Translation: web-18.0/web-18.0-web_timeline
Translate-URL: https://translation.odoo-community.org/projects/web-18-0/web-18-0-web_timeline/es/

Ed-Spain committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (10 of 10 strings)

Translation: web-18.0/web-18.0-web_time_range_menu_custom
Translate-URL: https://translation.odoo-community.org/projects/web-18-0/web-18-0-web_time_range_menu_custom/es/

Ed-Spain committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (18 of 18 strings)

Translation: web-18.0/web-18.0-web_company_color
Translate-URL: https://translation.odoo-community.org/projects/web-18-0/web-18-0-web_company_color/es/

Ed-Spain committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (21 of 21 strings)

Translation: web-18.0/web-18.0-web_pivot_computed_measure
Translate-URL: https://translation.odoo-community.org/projects/web-18-0/web-18-0-web_pivot_computed_measure/es/

Ed-Spain committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (20 of 20 strings)

Translation: web-18.0/web-18.0-web_ir_actions_act_multi
Translate-URL: https://translation.odoo-community.org/projects/web-18-0/web-18-0-web_ir_actions_act_multi/es/

Ed-Spain committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (42 of 42 strings)

Translation: web-18.0/web-18.0-web_quick_start_screen
Translate-URL: https://translation.odoo-community.org/projects/web-18-0/web-18-0-web_quick_start_screen/es/

OCA-git-bot committed:
[BOT] post-merge updates

Commit:
[UPD] Update web_chatter_position.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: web-18.0/web-18.0-web_chatter_position
Translate-URL: https://translation.odoo-community.org/projects/web-18-0/web-18-0-web_chatter_position/

shamnaskoyani committed:
[FIX] web_responsive: fix breadcrumb showing Unnamed on search nav

When navigating to a menu item via the apps menu search, the
breadcrumb displayed "Unnamed" instead of the actual action name.
This happened because search results used plain URL navigation
which bypasses the menu service and its action name context.

Use menuService.selectMenu() on search result click to navigate
through the proper menu service channel, ensuring the action
name is correctly passed to the breadcrumb component.

Closes OCA/web#3510

OCA-git-bot committed:
[BOT] post-merge updates

tarteo reviewed (commented):
**Functional test**

<img width="813" height="295" alt="Image" src="https://github.com/user-attachments/assets/4864792e-76e7-4f06-84ee-125fc4cd6228" />

is it possible to make it configurable also on mobile size screens?

<img width="622" height="196" alt="Image" src="https://github.com/user-attachments/assets/21eace4e-587a-4ece-8086-16892a19c555" />

Just my opinion, but would it make sense to remove “auto refresh” from the dropdown to save screen space?

For me the default button is a bit confusing, when I click it, it becomes the default for actions I haven't accessed yet only. I would expect that clicking the default button would save the current interval setting for the current action.
